logback-core/src/main/java/ch/qos/logback/core/rolling/helper/Compressor.java

```diff
@@ -45,37 +45,60 @@ public Compressor(CompressionMode compressionMode) {
   }
 
   /**
-   * @param nameOfFile2Compress
-   * @param nameOfCompressedFile
-   * @param innerEntryName       The name of the file within the zip file. Use for ZIP compression.
+   * Compresses a file and deletes it afterward if successful
+   * @param nameOfFile2Compress   File to be compressed
+   * @param nameOfCompressedFile  Destination filename
+   * @param innerEntryName        The name of the file within the zip file. Use for ZIP compression.
    */
   public void compress(String nameOfFile2Compress, String nameOfCompressedFile, String innerEntryName) {
-    switch (compressionMode) {
-      case GZ:
-        gzCompress(nameOfFile2Compress, nameOfCompressedFile);
-        break;
-      case ZIP:
-        zipCompress(nameOfFile2Compress, nameOfCompressedFile, innerEntryName);
-        break;
-      case NONE:
-        throw new UnsupportedOperationException(
-                "compress method called in NONE compression mode");
+    compress(nameOfFile2Compress, nameOfCompressedFile, innerEntryName, false);
+  }
+
+  /**
+   * Compresses a file and deletes it
+   * @param nameOfFile2Compress   File to be compressed
+   * @param nameOfCompressedFile  Destination filename
+   * @param innerEntryName        The name of the file within the zip file. Use for ZIP compression.
+   * @param alwaysDelete          Flag to force file deletion even if file couldn't be compressed
+   */
+  public void compress(String nameOfFile2Compress, String nameOfCompressedFile, String innerEntryName, boolean alwaysDelete) {
+    if (compressionMode == CompressionMode.NONE) {
+      throw new UnsupportedOperationException(
+          "compress method called in NONE compression mode");
+    }
+
+    boolean compressed = false;
+    try {
+      switch (compressionMode) {
+        case GZ:
+          compressed = gzCompress(nameOfFile2Compress, nameOfCompressedFile);
+          break;
+        case ZIP:
+          compressed = zipCompress(nameOfFile2Compress, nameOfCompressedFile, innerEntryName);
+          break;
+      }
+    } finally {
+      if (alwaysDelete || compressed) {
+        if (!new File(nameOfFile2Compress).delete()) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/tony19/logback/file/2371885200/issues/source?bid=3222618&fileBranchId=3222682#l82)